### ISSUE TITLE
Reactor selection in twisted.application should work on Py3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8299]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8299|
|Type|defect|
|Created|2016-04-17 06:53:33Z|
|Branch|https://github.com/twisted/twisted/tree/reactor-app-py3-8299|

Right now ```twisted/plugins/twisted_reactors.py``` isn't there, and so we don't get ```--reactor=foo``` working.

<details><summary>Searchable metadata</summary>

```
trac-id__8299 8299
type__defect defect
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_reactor_app_py3_8299 branches/reactor-app-py3-8299
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1460876013918614 1460876013918614
changetime__1460940943547336 1460940943547336
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [47275]) Branching to reactor-app-py3-8299.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Happy with this, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

can we have something like this

```
__all__ = ["default", "select", "poll", "epoll", "kqueue"]
if not _PY3:
   __all__.extend(["wx", "gi", "gtk2", "gtk3", "glib2", "glade"])
```

With the current approach I find it harder to see py2 has all py3 reactor + some more.

--------

For the release notes fragment
```
twisted.application-using applications (trial, twistd, etc)
```

Maybe

```
Twisted applications (trial, twistd, etc) using twisted.application framework ....
```

-------

All good.

ubuntu12.04-py2.7-coverage is failing, but I don't think that this is related

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [47288]) Merge reactor-app-py3-8299: Port twisted.plugins.twisted_reactors to Python 3

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8299](https://github.com/twisted/twisted/issues/8299)
